rest_framework/fields.py

```diff
@@ -653,6 +654,30 @@ def to_representation(self, value):
         return str(value)
 
 
+class IPAddressField(CharField):
+    """Support both IPAddressField and GenericIPAddressField"""
+
+    default_error_messages = {
+        'invalid': _('Enter a valid IPv4 or IPv6 address.'),
+    }
+
+    def __init__(self, protocol='both', unpack_ipv4=False, **kwargs):
+        self.protocol = protocol
+        self.unpack_ipv4 = unpack_ipv4
+        super(IPAddressField, self).__init__(**kwargs)
+        validators, error_message = ip_address_validators(protocol, unpack_ipv4)
+        self.validators.extend(validators)
+
+    def to_internal_value(self, data):
+        if data and ':' in data:
+            try:
+                return clean_ipv6_address(data, self.unpack_ipv4)
```

Comment:
I think this should be changed to:
```
data = clean_ipv6_address(data, self.unpack_ipv4)
```
or we just "clean" the ipv6 address without validating it.